### PR TITLE
Add the Tekton Documentation Contributor's Guide and Style Guide.

### PR DESCRIPTION
**Pull request by sergetron**
Adds the Tekton Documentation Contributor's Guide and Style Guide.

This PR just adds the relevant files. I will link to those files from CONTRIBUTE.md and other places in the Tekton docs in a separate PR.

### CODE REVIEW

**Comment by AlanGreene** [comment]
General question: do we have a preference for active over passive voice in the documentation?

**Comment by sergetron** [comment]
We should always use active. It's easier to parse and more user-friendly. I
think I mentioned it in the "Writing High Quality Documentation" doc and I
can also reiterate it in the "Content Guidelines" doc.

**Comment by AlanGreene** [comment]
I didn't see it called out explicitly so may have missed it. It's definitely included in the Google style guide under 'Language and grammar' (and in the [highlights](https://developers.google.com/style/highlights)). If we're pointing at the Google style guide my only concern is that it could be quite intimidating for newcomers if they feel they have to digest all of that before contributing docs.

So far the content is looking great :+1: I'll give it a proper read through this evening and add any other feedback.

**Comment by sergetron** [comment]
Just making sure, did you read all four files in the PR? The idea is we
cover all the most important things in our guide. The link to the
Google style guide is there if they want to go beyond what our guide covers.

File: content/en/doc-con-formatting.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    </tr>
+    <tr>
+     <td>Run the <code>smurfcaptor</code> command.
+     </td>
+     <td>Run smurfcaptor.
+     </td>
+    </tr>
+  </table>
+
+- Use angle brackets for placeholders, such as variables, and tell the reader what the placeholders represents. For example:
```
Comment (AlanGreene) [inline content/en/doc-con-formatting.md:98]:
```suggestion
- Use angle brackets for placeholders, such as variables, and tell the reader what the placeholders represent. For example:

```

File: content/en/doc-con-main.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+Once a pull request is created that addresses the issue, it goes through technical and editorial
+review, and is published on the Tekton documentation website.
+
+## Submitting a documentation change
+
+Submit a documentation change by opening a pull request against the relevant
+[Tekton repository](https://github.com/tektoncd). Documentation for each project is stored
+in the `docs` directory within the repository. 
+
+Tekton documentation is mirrored on the [Tekton website](https://www.tekton.dev) with the GitHub
```
Comment (AlanGreene) [inline content/en/doc-con-main.md:83]:
```suggestion
Tekton documentation is mirrored on the [Tekton website](https://tekton.dev) with the GitHub
```

the canonical URL omits `www.`

File: content/en/doc-con-writing.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+management? Aliens? Once you know your audience, you’ll know how much detail to include, the
+tone to use, and who your reviewers will be.
+
+Next, put yourself in your audience’s shoes and ask, “What do I, as this type of reader, need
+to know to use this thing I’m writing about?” Things that are obvious to you may not be obvious
+to an outsider. What tasks will the reader need to accomplish? What will they need to know to
+accomplish each task?
+
+Write down headings representing each of those major points or areas. For example, “Installation,”
+“Configuration,” “Operation modes,” or Command-line options.” These major things become the outline
```
Comment (AlanGreene) [inline content/en/doc-con-writing.md:48]:
```suggestion
“Configuration,” “Operation modes,” or “Command-line options.” These major things become the outline

```

File: content/en/doc-con-content.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+*   Break up long thoughts into multiple sentences.
+*   Break up longer documents into logical sections.
+*   Use paragraphs to separate thoughts and ideas.
+*   Ensure paragraphs flow into one another - that is, they are logically connected.
+*   Use headings and subheadings to indicate topics and sub-topics.
+
+## Style and tone
+
+- **Never use "should" or "might." **These words introduce uncertainty. We're either sure how our software works or we're not.
```
Comment (AlanGreene) [inline content/en/doc-con-content.md:39]:
```suggestion
- **Never use "should" or "might."** These words introduce uncertainty. We're either sure how our software works or we're not.
```

File: content/en/doc-con-content.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    </tr>
+    <tr>
+     <td>Version 3.0 adds support for Smurf counting.
+     </td>
+     <td>In version 3.0 we added Smurf counting support.
+     </td>
+    </tr>
+  </table>
+
+- **Never use "might" or "may." **"May" is often confused with "might" but they are not the same. "Might" introduces uncertainty just like "should," while "may" implies the permission has been granted and not that an option or action is available to the user. For example:
```
Comment (AlanGreene) [inline content/en/doc-con-content.md:110]:
```suggestion
- **Never use "might" or "may."** "May" is often confused with "might" but they are not the same. "Might" introduces uncertainty just like "should," while "may" implies the permission has been granted and not that an option or action is available to the user. For example:
```

File: content/en/doc-con-content.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+   </td>
+   <td><strong>Warning: </strong>Looking directly into the Basilisk's eyes turns you instantly to stone.
+   </td>
+  </tr>
+</table>
+
+## Things to avoid
+
+Avoid the following:
+- **Copy-pasting large portions of existing documentation.** If you need to reuse a large piece of content,
```
Comment (AlanGreene) [inline content/en/doc-con-content.md:167]:
```suggestion

- **Copy-pasting large portions of existing documentation.** If you need to reuse a large piece of content,
```
need a blank line before start of a list to ensure proper formatting

File: content/en/doc-con-formatting.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+---
+-->
+# Formatting conventions for Tekton documentation
+
+Tekton documentation uses Markdown to format the content. See the[Markdown cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
+to gain a basic understanding of Markdown conventions.
+
+## User interface elements
+
+- Use **bold **to indicate anything that's clickable, such as buttons or menu items.
```
Comment (AlanGreene) [inline content/en/doc-con-formatting.md:14]:
```suggestion
- Use **bold** to indicate anything that's clickable, such as buttons or menu items.
```

File: content/en/doc-con-formatting.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    </tr>
+  </table>
+
+- Remove trailing spaces. Some markdown objects, such as lists, require a specific number of spaces,
+  and often, the absence of spaces in specific places, in order to render properly on the website.
+  They can also cause unnecessary lint errors. 
+
+## Links
+
+- **Never use "here" or "this" as a link anchor. **Use either the exact title of the document being
```
Comment (AlanGreene) [inline content/en/doc-con-formatting.md:133]:
```suggestion
- **Never use "here" or "this" as a link anchor.** Use either the exact title of the document being
```

Comment (sergetron) [reply content/en/doc-con-formatting.md:133]: Fixed.

File: content/en/doc-con-formatting.md (hunk 7 above)
Comment (sergetron) [reply content/en/doc-con-formatting.md:14]: Fixed.

File: content/en/doc-con-content.md (hunk 6 above)
Comment (sergetron) [reply content/en/doc-con-content.md:167]: Fixed.

File: content/en/doc-con-content.md (hunk 5 above)
Comment (sergetron) [reply content/en/doc-con-content.md:110]: Fixed.

File: content/en/doc-con-content.md (hunk 4 above)
Comment (sergetron) [reply content/en/doc-con-content.md:39]: Fixed.

File: content/en/doc-con-writing.md (hunk 3 above)
Comment (sergetron) [reply content/en/doc-con-writing.md:48]: Fixed.

File: content/en/doc-con-main.md (hunk 2 above)
Comment (sergetron) [reply content/en/doc-con-main.md:83]: Didn't know that! Fixed.

File: content/en/doc-con-formatting.md (hunk 1 above)
Comment (sergetron) [reply content/en/doc-con-formatting.md:98]: Fixed.

**Comment by sergetron** [comment]
Thank you for the thorough review Alan! I added an explicit bullet about active vs. passive voice and fixed all the formatting issues. I'll be adding the new commits shortly.

**Comment by AlanGreene** [comment]
`sergetron@` should be `@sergetron` but let's get this in now and we can fix it up after since it's not actually linked from anywhere yet. I don't think GitHub usernames are automatically clickable on the website anyway, only when the markup is viewed in the GitHub UI.

/approve
/meow

**Pull request merged**